src/query/sql/src/planner/plans/insert.rs

```diff
@@ -82,6 +87,152 @@ impl Insert {
     pub fn has_select_plan(&self) -> bool {
         matches!(&self.source, InsertInputSource::SelectPlan(_))
     }
+
+    #[async_backtrace::framed]
+    pub async fn explain(
+        &self,
+        verbose: bool,
+    ) -> databend_common_exception::Result<Vec<DataBlock>> {
+        let mut result = vec![];
+
+        let Insert {
+            catalog,
+            database,
+            table,
+            schema,
+            overwrite,
+            // table_info only used create table as select.
+            table_info: _,
+            source,
+        } = self;
+
+        let table_name = format!("{}.{}.{}", catalog, database, table);
+        let inserted_columns = schema
+            .fields
+            .iter()
+            .map(|field| format!("{}.{} (#{})", table, field.name, field.column_id))
+            .collect::<Vec<_>>()
+            .join(",");
+
+        let mut children = vec![
+            FormatTreeNode::new(format!("table: {table_name}")),
+            FormatTreeNode::new(format!("inserted columns: [{inserted_columns}]")),
+            FormatTreeNode::new(format!("overwrite: {overwrite}")),
+        ];
+
+        let mut formatted_plan = String::new();
+
+        match source {
+            InsertInputSource::SelectPlan(plan) => {
+                if let Plan::Query {
+                    s_expr, metadata, ..
+                } = &**plan
+                {
+                    let metadata = &*metadata.read();
+                    let sub_tree = s_expr.to_format_tree(metadata, verbose)?;
```

Comment:
It’s would be better to display physical plan here, but due to the lack of `BindContext`, it is not easy to do it, I will change it to physical plan after refactoring Binder.

Reply:
Mark